PWGUD/TableProducer/UPCCandidateProducer.cxx

```diff
@@ -291,43 +291,48 @@ struct UpcCandProducer {
       }
     }
 
+    std::vector<int32_t> newMotherIDs{};
+
     // storing MC particles
     for (const auto& item : fNewPartIDs) {
       int32_t mcPartID = item.first;
       const auto& mcPart = mcParticles.iteratorAt(mcPartID);
       int32_t mcEventID = mcPart.mcCollisionId();
       int32_t newEventID = newEventIDs[mcEventID];
       // collecting new mother IDs
-      const auto& motherIDs = mcPart.mothersIds();
-      std::vector<int32_t> newMotherIDs;
-      newMotherIDs.reserve(motherIDs.size());
-      for (auto motherID : motherIDs) {
-        if (motherID >= nMCParticles) {
-          continue;
-        }
-        auto it = fNewPartIDs.find(motherID);
-        if (it != fNewPartIDs.end()) {
-          newMotherIDs.push_back(it->second);
+      if (mcPart.has_mothers()) {
+        const auto& motherIDs = mcPart.mothersIds();
+        for (auto motherID : motherIDs) {
+          if (motherID >= nMCParticles) {
+            continue;
+          }
+          auto it = fNewPartIDs.find(motherID);
+          if (it != fNewPartIDs.end()) {
+            newMotherIDs.push_back(it->second);
+          }
         }
       }
       // collecting new daughter IDs
-      const auto& daughterIDs = mcPart.daughtersIds();
       int32_t newDaughterIDs[2] = {-1, -1};
-      if (daughterIDs.size() > 0) {
-        int32_t firstDaughter = daughterIDs.front();
-        int32_t lastDaughter = daughterIDs.back();
-        if (firstDaughter >= nMCParticles || lastDaughter >= nMCParticles) {
-          continue;
-        }
-        auto itFirst = fNewPartIDs.find(firstDaughter);
-        auto itLast = fNewPartIDs.find(lastDaughter);
-        if (itFirst != fNewPartIDs.end() && itLast != fNewPartIDs.end()) {
-          newDaughterIDs[0] = fNewPartIDs.at(daughterIDs.front());
-          newDaughterIDs[1] = fNewPartIDs.at(daughterIDs.back());
+      if (mcPart.has_daughters()) {
+        const auto& daughterIDs = mcPart.daughtersIds();
+        if (daughterIDs.size() > 0) {
```

Comment:
Note that daughters index is a slice index, the ids always return `int[2]`, its size is always 2. It is enough to check with `has_daughters()` that checks for the both numbers to be non-negative.

Reply:
Thank you, Anton!